apps/kyberswap-interface/src/components/Web3Provider/SolanaProvider.tsx

```diff
@@ -0,0 +1,89 @@
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token'
+import { ConnectionProvider, WalletProvider, useConnection, useWallet } from '@solana/wallet-adapter-react'
+// import { WalletAdapterNetwork } from '@solana/wallet-adapter-base'
+import { WalletModalProvider } from '@solana/wallet-adapter-react-ui'
+// import { clusterApiUrl } from '@solana/web3.js'
+import '@solana/wallet-adapter-react-ui/styles.css'
+import { FC, ReactNode, createContext, useContext, useEffect, useState } from 'react'
+
+import { SOLANA_RPC } from 'constants/env'
+
+interface SolanaProviderProps {
+  children: ReactNode
+}
+
+export const SolanaProvider: FC<SolanaProviderProps> = ({ children }) => {
+  // const network = WalletAdapterNetwork.Mainnet
+
+  // You can also provide a custom RPC endpoint
+  // const endpoint = useMemo(() => clusterApiUrl(network), [network])
+
+  return (
+    <ConnectionProvider endpoint={SOLANA_RPC}>
+      <WalletProvider wallets={[]} autoConnect>
+        <WalletModalProvider>
+          <SolanaTokenBalances>{children}</SolanaTokenBalances>
+        </WalletModalProvider>
+      </WalletProvider>
+    </ConnectionProvider>
+  )
+}
+
+interface SolanaTokenBalance {
+  mint: string
+  balance: number
+  decimals: number
+  rawAmount: string
+}
+
+const SolanaTokenBalanceContext = createContext<Record<string, SolanaTokenBalance> | null>(null)
+
+const SolanaTokenBalances: FC<SolanaProviderProps> = ({ children }) => {
+  const [balances, setBalances] = useState<Record<string, SolanaTokenBalance>>({})
+  const { publicKey: solanaAddress } = useWallet()
+  const connection = useConnection()
+
+  useEffect(() => {
+    if (!solanaAddress) return
+    const fetchBalance = async () => {
+      const nativeBalance = await connection.connection.getBalance(solanaAddress)
+      const tokenAccounts = await connection.connection.getParsedTokenAccountsByOwner(solanaAddress, {
+        programId: TOKEN_PROGRAM_ID,
+      })
+      const b = tokenAccounts.value
+        .map(account => {
+          const info = account.account.data.parsed.info
+          return {
+            mint: info.mint,
+            balance: info.tokenAmount.uiAmount,
+            decimals: info.tokenAmount.decimals,
+            rawAmount: info.tokenAmount.amount,
+          }
+        })
+        .reduce((acc, curr) => ({ ...acc, [curr.mint]: curr }), {} as Record<string, SolanaTokenBalance>)
+      setBalances({
+        '11111111111111111111111111111111': {
+          mint: '11111111111111111111111111111111',
```

Comment:
Should we use `export const SOLANA_NATIVE = '11111111111111111111111111111111'`?